acctz/acctz.proto

```diff
@@ -132,6 +132,13 @@ message UserDetail {
   // which is implementation dependent.  Might be omitted if
   // authentication fails.
   string group = 2;
+
+  // this indicates the role of the user.
+  string role = 3;
+
+  // this indicates the principal present in SSH certificates populated
+  // when the user is authenticated via SSH certificates.
+  string principal = 4;
```

Comment:
If i am reading this right in this case
ssh cert principal (hines) ssh login root@

identity would be "root"
principal would be "hines"

ssh without cert "hines"
identity would be "hines"
principal unset


ssh without cert "root"
identity would be "root"
principal unset

I get this however it feels generally counter to what the acccounting record is about.
The identity we care about for accounting is ALWAYS the most specific authorization user (ssh user name if no cert or cert identity if present)

it is slightly more annoying to evaluate the records directly if you have to have a "if check" to determine the actual user

When developing this identity field that was the intent to encode the ACTUAL end identity
The "principal" really would end up being more of "login-user" or "effective-user-id"

Reply:
Operationally, if we only have "the most specific authorization user", let's say the principal on a certificate, we don't have enough information about what identity hat most specific user is using. (hines / root) != (hines / readonly) from an accounting perspective at a minimum. I don't love overloading fields, and a principal on a certificate and "the thing before the @", are distinct useful pieces of accounting data. Seeing all the confused discussion on these things that keeps coming up, it probably does serve us well to optimize the naming for most clear understanding.

Reply:
I think we agree that we would need `the thing before the @` and the principal. `the thing before the @` is called `identity` in the proto. We can name it different if it doesn't make sense but maybe in a different PR.

`role` field which is being added in this PR is not the name of a key in a hiba grant but a free form string like `network-operator`, `network-admin` etc. There was discussions about using the `group` field to represent the `role`, `privilege-level`, `task-group` etc but some NOS have multiple information so below is my proposal,

1. Name the field principal as `ssh-principal` so that its clear.
2. Remove the proposed `role` field.
3. Have the field `group` be a map with the key as `role`/ `privilege-level` / `task-group` so that multiple values can be represented.

@sourcequench @haussli @marcushines @morrowc

Reply:
+1 for clarity, and `ssh-principal` or another not-likely-to-be-overloaded-or-confused field is better than `identity` to me. I have less of an opinion on capturing these sometimes vendor-specific bits of security context in the group field - although I don't like the name group.

Reply:
From a side discussion with @morrowc, we looked for other cases of 'group' (priv level, or whatever you might call it) in other gnxi models.  Not an extensive search, but there is precedence for calling this 'role'.  IMO, acctz should be changed to be consistent with this; ie: 'group' -> 'role':

https://github.com/openconfig/public/blob/master/release/models/system/openconfig-aaa.yang#L398-L411
https://github.com/openconfig/public/blob/master/release/models/system/openconfig-aaa-types.yang#L77

Reply:
We are leaning towards `group` instead of `role` since role is also a group. I have updated the comment for `group` to include the user role based on the implementation.

Reply:
Updated the filed to role